<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>468</width>
    <height>620</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Password Generator App</string>
  </property>
  <property name="styleSheet">
   <string notr="true">         QWidget {
                background-color: #18171F;
                color: #D8DEE9;
                font-family:Inter;
                font-size: 11pt;
            }</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="passwordLengthLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>130</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Password Length</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="generatedPasswordLineEdit">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>8</y>
      <width>391</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color: #3B4252;
color:  #E6E6E6;
border-color: #3B4252;</string>
    </property>
    <property name="frame">
     <bool>false</bool>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="copyPushButton">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>8</y>
      <width>50</width>
      <height>61</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QPushButton{background-color:#18171F;
border-radius:30px
}</string>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset>
      <normaloff>assets/icons/paste64px.png</normaloff>assets/icons/paste64px.png</iconset>
    </property>
    <property name="iconSize">
     <size>
      <width>32</width>
      <height>32</height>
     </size>
    </property>
    <property name="checkable">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QSlider" name="passwordLengthSlider">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>150</y>
      <width>450</width>
      <height>61</height>
     </rect>
    </property>
    <property name="styleSheet">
     <string notr="true">QSlider::groove:horizontal{
	border: 2px solid #3B4252;
	height: 4px;
	border-radius: 5
}
			

QSlider::handle:horizontal{
	background: #E6E6E6;
	width : 20px;
	margin: -10px -2px;
	border-radius: 11px;
	border: 2px solid #2a2a2a;
    
}


Qslider::add-page:horizontal {
background: qlineargradient(x1:0, y1:0,x2:0, y2:1, stop:0 #B1B1B1, stop:1 
grey);

}

QSlider::sub-page:horizontal {
background: #A4FFAF ;

}


QSlider::handle:horizontal:hover{
	background:#18171F;
	border-color:#A4FFAF;
}</string>
    </property>
    <property name="minimum">
     <number>8</number>
    </property>
    <property name="maximum">
     <number>64</number>
    </property>
    <property name="sliderPosition">
     <number>12</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
    <property name="tickPosition">
     <enum>QSlider::TicksAbove</enum>
    </property>
    <property name="tickInterval">
     <number>0</number>
    </property>
   </widget>
   <widget class="QCheckBox" name="uppercaseCheckBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>230</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox{
	spacing:10px
}

QCheckBox::indicator:checked {
	image: url(:/checkbox_icon/assets/icons/check(2).png);
}
QCheckBox::indicator:unchecked {
	
	image: url(:/checkbox_icon/assets/icons/unchecked.png);
}
</string>
    </property>
    <property name="text">
     <string>Include Uppercase Letters (A-Z)</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
    <property name="tristate">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="lowercaseCheckBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>280</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox{
	spacing:10px
}

QCheckBox::indicator:checked {
	image: url(:/checkbox_icon/assets/icons/check(2).png);
}
QCheckBox::indicator:unchecked {
	
	image: url(:/checkbox_icon/assets/icons/unchecked.png);
}
</string>
    </property>
    <property name="text">
     <string>Include Lowercase Letters (a-z)</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="symbolCheckBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>380</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox{
	spacing:10px
}

QCheckBox::indicator:checked {
	image: url(:/checkbox_icon/assets/icons/check(2).png);
}
QCheckBox::indicator:unchecked {
	
	image: url(:/checkbox_icon/assets/icons/unchecked.png);
}
</string>
    </property>
    <property name="text">
     <string>Include Symbols (!&quot;#$%&amp;'()*...)</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QCheckBox" name="numbersCheckBox">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>330</y>
      <width>301</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">QCheckBox{
	spacing:10px
}

QCheckBox::indicator:checked {
	image: url(:/checkbox_icon/assets/icons/check(2).png);
}
QCheckBox::indicator:unchecked {
	
	image: url(:/checkbox_icon/assets/icons/unchecked.png);
}
</string>
    </property>
    <property name="text">
     <string>Include Numbers (0-9)</string>
    </property>
    <property name="checked">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QProgressBar" name="passwordStrengthProgressBar">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>445</y>
      <width>291</width>
      <height>21</height>
     </rect>
    </property>
    <property name="value">
     <number>50</number>
    </property>
    <property name="textVisible">
     <bool>false</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="generatePushButton">
    <property name="geometry">
     <rect>
      <x>36</x>
      <y>510</y>
      <width>391</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">background-color:#A4FFAF;
color: #18171F;</string>
    </property>
    <property name="text">
     <string>GENERATE  </string>
    </property>
   </widget>
   <widget class="QLabel" name="passwordStrengthLabel">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>440</y>
      <width>101</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">color:lightgrey;</string>
    </property>
    <property name="text">
     <string>STRENGHT :</string>
    </property>
   </widget>
   <widget class="QLabel" name="sliderValueLabel">
    <property name="geometry">
     <rect>
      <x>409</x>
      <y>90</y>
      <width>51</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Inter</family>
      <pointsize>11</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>12</string>
    </property>
    <property name="alignment">
     <set>Qt::AlignCenter</set>
    </property>
   </widget>
   <zorder>copyPushButton</zorder>
   <zorder>passwordLengthLabel</zorder>
   <zorder>generatedPasswordLineEdit</zorder>
   <zorder>passwordLengthSlider</zorder>
   <zorder>uppercaseCheckBox</zorder>
   <zorder>lowercaseCheckBox</zorder>
   <zorder>symbolCheckBox</zorder>
   <zorder>numbersCheckBox</zorder>
   <zorder>passwordStrengthProgressBar</zorder>
   <zorder>generatePushButton</zorder>
   <zorder>passwordStrengthLabel</zorder>
   <zorder>sliderValueLabel</zorder>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>468</width>
     <height>23</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
